src/libcore/str/mod.rs

```diff
@@ -2501,6 +2505,21 @@ impl StrExt for str {
 
     #[inline]
     fn parse<T: FromStr>(&self) -> Result<T, T::Err> { FromStr::from_str(self) }
+
+    #[inline]
+    fn split_off<'a>(self: &mut &'a Self, at: usize) -> &'a str {
+        let (rest, split) = self.split_at(at);
+        *self = rest;
+        split
+    }
+
+    #[inline]
+    fn split_off_mut<'a>(self: &mut &'a mut Self, at: usize) -> &'a mut str {
+        let (rest, split) = mem::replace(self, unsafe { from_utf8_unchecked_mut(&mut []) })
```

Comment:
This is trivially safe since an empty string slice is UTF-8 by definition, but a comment to that effect would be nice here ;)
(Generally I think all uses of unsafe elimination blocks should come with a comment explaining the reason why it is safe.)

Reply:
I finally added the damn comment. :3